### ISSUE TITLE
[Bug]: Unexpected behavior with `belowLayerID` on Android

### ISSUE THREAD

**Issue by grantsisson** [rnmapbox/maps#3572]
### Mapbox Implementation

Mapbox

### Mapbox Version

default

### React Native Version

0.74.3

### Platform

Android

### `@rnmapbox/maps` version

10.1.27

### Standalone component to reproduce

```ts
import React, { ComponentProps } from "react";
import { Button, StyleProp } from "react-native";
import Mapbox, {
  MapView,
  ShapeSource,
  CircleLayer,
  Camera,
  CircleLayerStyle,
  SymbolLayerStyle,
  SymbolLayer,
} from "@rnmapbox/maps";

const CAMERA_CENTER = [-74.007, 40.7243];

const circleLayerStyles: StyleProp<CircleLayerStyle> = {
  layer1: { circleColor: "#f19797", circleRadius: 40 },
  layer2: { circleColor: "#a6b5f2", circleRadius: 30 },
};

const symbolLayerStyles: StyleProp<SymbolLayerStyle> = {
  textField: ["get", "title"],
  textSize: 14,
  textColor: "black",
  textAnchor: "center",
  textAllowOverlap: true,
};

type Shape = ComponentProps<typeof ShapeSource>["shape"];

const features: Shape = {
  type: "FeatureCollection",
  features: [
    {
      type: "Feature",
      id: "a-feature",
      properties: { title: "Circle A" },
      geometry: { type: "Point", coordinates: CAMERA_CENTER },
    },
  ],
};

const features2: Shape = {
  type: "FeatureCollection",
  features: [
    {
      type: "Feature",
      id: "b-feature",
      properties: { title: "Circle B" },
      geometry: {
        type: "Point",
        coordinates: [CAMERA_CENTER[0], CAMERA_CENTER[1] + 0.02],
      },
    },
  ],
};

function BugReportExample() {
  const [belowLayerID, setBelowLayerID] = React.useState<string>();
  return (
    <>
      <MapView style={{ flex: 1 }}>
        <Camera
          defaultSettings={{ centerCoordinate: CAMERA_CENTER, zoomLevel: 10 }}
        />
        <ShapeSource id="shape-source-id-1" shape={features}>
          <CircleLayer id="circle-layer-1" style={circleLayerStyles.layer1} />
          <SymbolLayer id="symbol-layer-1" style={symbolLayerStyles} />
        </ShapeSource>
        <ShapeSource id="shape-source-id-2" shape={features2}>
          <CircleLayer
            id="circle-layer-2"
            style={circleLayerStyles.layer2}
            belowLayerID={belowLayerID}
          />
          <SymbolLayer id="symbol-layer-2" style={symbolLayerStyles} />
        </ShapeSource>
      </MapView>
      <Button
        title={`Toggle below layer ID prop: ${belowLayerID ?? "undefined"}`}
        onPress={() =>
          setBelowLayerID((prev) => (prev ? undefined : "circle-layer-1"))
        }
      />
    </>
  );
}

export default BugReportExample;

```

### Observed behavior and steps to reproduce

On Android, when I set the `belowLayerID` prop for a given layer, the layer disappears from the map. However, this works as expected on iOS: the rendered order changes.

Steps to reproduce:

1. Render two `ShapeSource`s each with a `CircleLayer` child and at least one relevant feature.
2. Set the `belowLayerID` for one of the `CircleLayer`s to the `id` of the other. Hopefully clear in the component code provided, but lmk if not!
3. The `CircleLayer` with the `belowLayerID` specified disappears from the map.

Here's an example with the standalone component provided:

![Jul-15-2024 16-52-04](https://github.com/user-attachments/assets/5a21ee9d-8295-48e4-ad59-4db89ba0caae)

### Expected behavior

I'm expecting the layer with `belowLayerID` set to render underneath the other (analogous to changing the z-index on html elements). This works as expected on iOS.

iOS with the standalone component provided:

![Jul-15-2024 16-55-27](https://github.com/user-attachments/assets/1404786a-2716-43a5-ba27-5689e8ea70db)

### Notes / preliminary analysis

Maybe worth noting that, on both platforms, toggling the `beforeLayerID` back to `undefined` has no effect on the rendered layers (meaning, Circle B will remain beneath Circle A). The gifs are slightly deceptive in that regard.

Also, this example was created with Expo 50. If helpful, happy to provide an example project. If I'm misunderstanding how to use the API here, please let me know. Thanks for the help!

**Comment by wyattberlinic** [comment]
I think I see the bug here.

In [setBelowLayerID](https://github.com/rnmapbox/maps/blob/25784b67ef5963456208d2d31d5cf7764d8550e3/android/src/main/java/com/rnmapbox/rnmbx/components/styles/layers/RNMBXLayer.kt#L70) you call two functions:
https://github.com/rnmapbox/maps/blob/25784b67ef5963456208d2d31d5cf7764d8550e3/android/src/main/java/com/rnmapbox/rnmbx/components/styles/layers/RNMBXLayer.kt#L79-L80

`removeFromMap` sets `mLayer = null`.
https://github.com/rnmapbox/maps/blob/25784b67ef5963456208d2d31d5cf7764d8550e3/android/src/main/java/com/rnmapbox/rnmbx/components/styles/layers/RNMBXLayer.kt#L310-L312

But then in `addBelow` the callback does nothing if not initialized:
https://github.com/rnmapbox/maps/blob/25784b67ef5963456208d2d31d5cf7764d8550e3/android/src/main/java/com/rnmapbox/rnmbx/components/styles/layers/RNMBXLayer.kt#L190-L192

`hasInitialized` returns `False` if `mLayer == null`:
https://github.com/rnmapbox/maps/blob/25784b67ef5963456208d2d31d5cf7764d8550e3/android/src/main/java/com/rnmapbox/rnmbx/components/styles/layers/RNMBXLayer.kt#L328-L330

So TLDR:
1. `removeFromMap` sets `mLayer = null`.
2. `addBelow` does nothing if `mLayer == null`.

**Comment by mfazekas** [comment]
@wyattberlinic amazing, thanks much for looking into it.

I think the issue is:
https://github.com/rnmapbox/maps/blob/25784b67ef5963456208d2d31d5cf7764d8550e3/android/src/main/java/com/rnmapbox/rnmbx/components/styles/layers/RNMBXLayer.kt#L310-L312

As per https://github.com/rnmapbox/maps/pull/3392  this is to force recreation of mLayer when the underlying style changes, this is good and necessary. But with `RemoveReason.REORDER` we don't need to set `mLayer = null` 

@wyattberlinic What do you think?

**Comment by wyattberlinic** [comment]
Yes, that makes sense to me. This is my first time looking at this codebase but I'd expect that to work as we're re-applying the style immediately in `addBelow` or `addAbove`.

**Comment by grantsisson** [comment]
Thanks both. I have this project mostly set up now so happy to attempt this change and to validate it. Probably not until later this week/this weekend though.
